google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -971,37 +983,46 @@ public long executeUpdate(Statement update) {
     Preconditions.checkNotNull(update);
     ConnectionPreconditions.checkState(!isClosed(), CLOSED_ERROR_MSG);
     ParsedStatement parsedStatement = getStatementParser().parse(update);
-    if (parsedStatement.isUpdate()) {
-      switch (parsedStatement.getType()) {
-        case UPDATE:
+    switch (parsedStatement.getType()) {
+      case UPDATE:
+        if (this.autocommitDmlMode == AutocommitDmlMode.PARTITIONED_NON_ATOMIC) {
           return get(internalExecuteUpdateAsync(parsedStatement));
-        case CLIENT_SIDE:
-        case QUERY:
-        case DDL:
-        case UNKNOWN:
-        default:
-      }
+        }
+        ResultSet rs = internalExecuteQuery(parsedStatement, AnalyzeMode.NONE);
+        if (rs.getColumnCount() > 0) {
+          throw SpannerExceptionFactory.newSpannerException(
+              ErrorCode.INVALID_ARGUMENT,
+              "Statement is not an normal DML statement: "
+                  + parsedStatement.getSqlWithoutComments());
+        }
+        while (rs.next()) ;
```

Comment:
We should add a comment to explain why we are doing this. Also, according to the Google style guide, a `while` statement must have curly braces.

Reply:
Done.